api4/command.go

```diff
@@ -212,7 +212,13 @@ func executeCommand(c *Context, w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	commandArgs.TeamId = channel.TeamId
+	if commandArgs.TeamId == "" {
+		commandArgs.TeamId = channel.TeamId
+	} else if c.Session.GetTeamByTeamId(commandArgs.TeamId) == nil {
+		c.SetPermissionError(model.PERMISSION_USE_SLASH_COMMANDS)
```

Comment:
The v3 behavior is to return a permission error. I don't really have a strong opinion on the error code being 400 vs 403 in this case, so I left it as it was in v3.